### ISSUE TITLE
Incorrect document.domain

### ISSUE THREAD

**Issue by ghengeveld** [cypress-io/cypress#2193]
### Is this a Feature or Bug?

Bug.

### Current behavior:

When running Cypress against a subdomain (e.g. sub.example.com), `document.domain` is reported as `example.com`.

### Desired behavior:

`document.domain` should be the full domain name `sub.example.com`.

### Steps to reproduce:

Run a Cypress test against a subdomain somewhere, then run `document.domain` in the dev console.

### Versions

Latest Cypress using Chrome Canary.

**Comment by brian-mann** [comment]
This is the correct behavior. I'm not sure why you're asking for it to be different. Setting it to the superdomain is what is intended so that subdomains do not break origin policy. Otherwise they would be considered cross origin frames.

**Comment by ghengeveld** [comment]
It's inconsistent with normal browser behavior. In my case it breaks my app. Luckily I can emulate it using `location.href` instead:
```
const domain = document.location.href.match(/\/\/([^:/]+)/)[1]
```
Since I have a workaround, I can live with the inconsistency, but maybe it's worth reconsidering this behavior.

**Comment by jennifer-shehane** [comment]
I also do not believe this to be correct behavior. If you manually visit "https://en.m.wikipedia.org/wiki/Main_Page" and do `document.domain` in the console - it prints `en.m.wikipedia.org` which is NOT what it returns within Cypress.

**Failing reproducible example:**
```js
it('document.domain not right', function () {
  cy.visit('https://en.m.wikipedia.org/wiki/Main_Page')
  cy.document().its('domain').should('eq', 'en.m.wikipedia.org')
})
```

**Comment by turkeyzawa** [comment]
Is there any progress being made on this issue? This behavior makes our app broken because we use `document.domain` in the javascript to call our backend's apis.

**Comment by jennifer-shehane** [comment]
This issue is still in the 'ready for work' stage, which means no work has been done on this issue as of today, so we do not have an estimate on when this will be delivered.

**Comment by nicojs** [comment]
@jennifer-shehane are you open to a PR for this issue? If so, could you point potential first time committers in the right direction? 

I would propose being able to override the default behavior of using the "superdomain", instead letting users use the sub-domain and thus accepting the limitation of not being able to visit other (sub-)domains.

**Comment by tdelphi** [comment]
I also run into this bug.
In my case – my app is using OWASP CSRFGuard with pretty strict configuration and JS protection. Which blocks every request that does look suspicious. 
So, when my app is running on internal domain https://myPc.domain1.domain2.com but cypress changes each document.domain to domain2.com, then OWASP CSRFGuard stops injecting CSRFToken into HTML forms, and after that every single request leads to “we are getting hacked” page of my app.
As a possible solution - it would be sufficient for me to be able to hard-code desired domain inside tests.

**Comment by nicojs** [comment]
Yeah, i think document.domain is a pretty big side effect. I'm trying for a couple of days to make it work but simply cannot. However, I don't feel inclined to fix it without an "approval" of a Cypress core contributor.

**Comment by nicojs** [comment]
Thanks for your response.

> @nicojs I'm not sure about the issue you are encountering exactly, since you never described it, so let us know if there is some limitations to our suggestion.

Our (single page) application is loaded from a CMS in a big enterprise company. It also loads a lot of 3rd party JavaScript being loaded from different domains. The 3rd party scripts are used to load a chat application, a cookie consent form, stuff like that. The 3rd party JavaScript determines which additional JavaScript to load dynamically from the main domain. The only way it can load more JavaScript is by adding `script` tags to the page with the `src` attribute pointing to the original domain. It does that by, you guessed it, using the infamous `document.domain`. It results in invalid urls which in turn makes it so that the page load event never occurs. Our test script fails with a timeout because the `visit` never finishes. We don't have direct control over the 3rd party scripts. I would like to see it differently, but we cannot change the way the entire organization works. I've seen these practices in more enterprise organizations, so I don't feel like it is uncommon.

> With the way Cypress works today, we have to set the document.domain in this way to avoid issues with visiting other subdomains. 

Like I stated earlier: I can understand this reasoning, however we don't need this feature in our use case. That's why I was proposing making this behavior optional.

**Comment by sscaff1** [comment]
We have the same situation with a very old application. This is preventing us from using cypress for testing...We were hoping to move from Selenium to Cypress...

Why can't this be configurable?

**Comment by DanielPhangHL** [comment]
Hi is there any update on how to overcome this problem without turning off the OWASP CSRFGuard restrictions during testing ? Please reply as our company QA team is encountering this issue.

**Comment by DanielPhangHL** [comment]
I see that @nagash77 move the issue to Closed. Any update is this problem fixed ? Please answer.

**Comment by nagash77** [comment]
@DanielPhangHL Issues in our GitHub repo are reserved for potential bugs or feature requests. 

We recommend questions relating to how to use Cypress be asked in our [community chat](https://discord.gg/cypress). Also try searching our [existing GitHub issues](https://gitter.im/cypress-io/cypress/issues), reading through our [documentation](https://docs.cypress.io), or searching [Stack Overflow](https://stackoverflow.com/questions/tagged/cypress) for relevant answers.
